Java benchmarks broken with recent versions of Java SDK (only works with JDK 8)

Based on my experiments (https://github.com/protocolbuffers/protobuf/pull/8665), the java benchmark from https://github.com/protocolbuffers/protobuf/blob/0b8d13a1d4cd9be16ed8a2230577aa9c296aa1ca/kokoro/linux/benchmark/run.sh#L83 only works when JDK 8 is used. On newer JDKs (I tried openjdk 11), the benchmark scenarios start, but they end with an error and don't report any results.

The reported error:
```
This selection yields 4 experiments.
ERROR: Trial failed to complete (its results will not be included in the run):
  The worker exited without producing data. It has likely crashed. Inspect /tmp/1622479310741-0/trial-1.log to see any worker output.
```

the trial-1.log contains
```
...
[stdout]      42    1    b        java.lang.StringLatin1::hashCode (42 bytes)
[stdout]      73    2     n       java.lang.invoke.MethodHandle::linkToStatic(LLLLLLL)L (native)   (static)
[stdout]      75    3     n       java.lang.invoke.MethodHandle::linkToSpecial(LL)L (native)   (static)
[stdout]      76    4     n       java.lang.invoke.MethodHandle::invokeBasic(LLLLLL)L (native)   
[stdout]      76    5     n       java.lang.invoke.MethodHandle::linkToSpecial(LLLLLLLL)L (native)   (static)
[stdout]      79    6     n       java.lang.invoke.MethodHandle::linkToStatic(LL)L (native)   (static)
[stdout]      80    7     n       java.lang.invoke.MethodHandle::invokeBasic(L)L (native)   
[stdout]      80    8     n       java.lang.invoke.MethodHandle::linkToSpecial(LLL)L (native)   (static)
[stdout]      81    9     n       java.lang.invoke.MethodHandle::linkToStatic(LLL)L (native)   (static)
[stdout]      82   10     n       java.lang.invoke.MethodHandle::invokeBasic(LL)L (native)   
[stdout]      82   11     n       java.lang.invoke.MethodHandle::linkToSpecial(LLLL)L (native)   (static)
[stdout]      85   12    b        java.util.Properties$LineReader::readLine (635 bytes)
[stdout]      99   12             java.util.Properties$LineReader::readLine (635 bytes)   made not entrant
[stdout]     100   13 %  b        java.util.Properties$LineReader::readLine @ 224 (635 bytes)
[stdout]     114   14    b        java.util.Properties$LineReader::readLine (635 bytes)
[stdout]     122   14             java.util.Properties$LineReader::readLine (635 bytes)   made not entrant
[stdout]     126   15     n       java.lang.invoke.MethodHandle::linkToStatic(LLLL)L (native)   (static)
[stdout]     127   16     n       java.lang.invoke.MethodHandle::invokeBasic(LLL)L (native)   
[stdout]     127   17     n       java.lang.invoke.MethodHandle::linkToSpecial(LLLLL)L (native)   (static)
[stdout]     134   18    b        java.lang.String::isLatin1 (19 bytes)
[stderr] Error: Could not find or load main class com.google.caliper.worker.WorkerMain
[stderr] Caused by: java.lang.ClassNotFoundException: com.google.caliper.worker.WorkerMain
```

--- a/benchmarks/java/pom.xml
+++ b/benchmarks/java/pom.xml
@@ -22,7 +22,7 @@
     <dependency>
       <groupId>com.google.caliper</groupId>
       <artifactId>caliper</artifactId>
-      <version>1.0-beta-2</version>
+      <version>1.0-beta-3</version>
     </dependency>
   </dependencies>
 

